mark sshPublicKey as optional, docs windows ssh (#3241)
kubernetes-sigs/cluster-api-provider-azure: docs/book/src/topics/windows.md

### VM password and access
The VM password is [random generated](https://cloudbase-init.readthedocs.io/en/latest/plugins.html#setting-password-main)

> by Cloudbase-init during provisioning of the VM. For Access to the VM you can use ssh which can be configured with SSH

Reviewer: ```suggestion
by Cloudbase-init during provisioning of the VM. For Access to the VM you can use ssh, which can be configured with a
```

public key you provide during deployment. For example like this:
```yaml
apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
kind: KubeadmConfigTemplate
metadata:
  name: test1-md-0
  namespace: default
spec:
  template:
    spec:
      ...
      users:
      - name: username
        groups: Administrators
        sshAuthorizedKeys:
        - "ssh-rsa AAAA..."
```

> Refer to [SSH Access for nodes](ssh-access.md) for instructions on how to add an SSH public key for Windows instances.

Reviewer 2: Should this be "for Linux Instances"?

Author: I think that sentence is wrong/outdaterd since `ssh-access.md` seems to talk about how to connect to nodes and not about how to add SSH public keys. Will adjust.

> Refer to [SSH Access for nodes](ssh-access.md) for instructions on how to add an SSH public key for windows instances.

Reviewer: ```suggestion
Refer to [SSH Access for nodes](ssh-access.md) for instructions on how to add an SSH public key for Windows instances.
```

> It's required to specify the SSH key using the `users` property in the Kubeadm config template. Specifying the `sshPublicKey` on `AzureMachine` / `AzureMachinePool` resources only works with Linux instances.

Reviewer 2: Should this be up by line 105?

Author: I put it here because the sentence right before talks about how to add SSH access for windows nodes.

> It's required required to specify the ssh key using the `users` property in the Kubeadm config template. Specifying the `sshPublicKey` on `AzureMachine` / `AzureMachinePool` resources does only work with Linux instances.

Reviewer: ```suggestion
It's required to specify the SSH key using the `users` property in the Kubeadm config template. Specifying the `sshPublicKey` on `AzureMachine` / `AzureMachinePool` resources only works with Linux instances.
```

> Ensure you use a different name than `capi` for the user (see [issue #3232](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3232) for details). 

Reviewer 2: This should be fixed with https://github.com/kubernetes-sigs/image-builder/pull/1109
